packages/lib-utils/src/k8s/hooks/usePrevious.ts

```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+
+export const usePrevious = <P = unknown>(value: P, deps?: unknown[]): P | undefined => {
```

Comment:
I wonder if we could also improve our types here - I don't want the shared `usePrevious` utility to be a reason why we do not focus on explicit types in components/hooks.

Reply:
@florkbr I've added this as an additional point in the Acceptance Criteria for the ticket here: https://issues.redhat.com/browse/HAC-801